From a548f85913a42fe1edab4333fb4b3d02c2680b55 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Sun, 25 Jul 2021 10:47:26 +0300
Subject: [PATCH] [SW-45031] greco: remove duplicate code

Unified and moved reset specific functions
implemented for greco_arm and greco_arc
to greco/common/reset.c file.

Change-Id: Iabeaae5b8d768a0724a152786932b7778ad681a3
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/arch/greco/arc/Makefile b/arch/greco/arc/Makefile
index 9cf3021..a32aa6f 100644
--- a/arch/greco/arc/Makefile
+++ b/arch/greco/arc/Makefile
@@ -3,4 +3,3 @@
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/pll.c)
 PREBOOT_SRCS_C += $(shell ls lib/preboot/thermal_v2.c)
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/security.c)
-PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/reset.c)
diff --git a/arch/greco/arc/common.h b/arch/greco/arc/common.h
index a9e85bd..c3e3e96 100644
--- a/arch/greco/arc/common.h
+++ b/arch/greco/arc/common.h
@@ -54,4 +54,8 @@
 
 #define FW_WORD_NUM		0x2200	/* FW_SIZE / 2 */
 
+#define SW_ALL_RST_L_MSK		0xC00FF8FF
+#define SW_ALL_RST_M_MSK		0xFFFFFFFF
+#define SW_ALL_RST_H_MSK		0x7FFFFF
+
 #endif /* COMMON_GRECO_ARC_GNU_H_ */
diff --git a/arch/greco/arc/reset.c b/arch/greco/arc/reset.c
deleted file mode 100644
index 70a8aa8..0000000
--- a/arch/greco/arc/reset.c
+++ /dev/null
@@ -1,41 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0
-/*
- * Copyright 2021 HabanaLabs, Ltd.
- * All Rights Reserved.
- */
-
-#include <psoc_global_conf_regs.h>
-#include <psoc_global_conf_masks.h>
-
-#include <bitops.h>
-#include <preboot.h>
-#include <pll.h>
-#include <uart.h>
-
-#include "common.h"
-
-#define SW_ALL_RST_L_MSK		0xC00FF8FF
-#define SW_ALL_RST_M_MSK		0xFFFFFFFF
-#define SW_ALL_RST_H_MSK		0x7FFFFF
-
-static void greco_hw_fini(void)
-{
-	/* Restart BTL/BLR upon hard-reset */
-	regmap_writel(0x770, mmPSOC_GLOBAL_CONF_BOOT_SEQ_RE_START);
-
-	udelay(1000);
-
-	regmap_writel(SW_ALL_RST_L_MSK, mmPSOC_GLOBAL_CONF_SW_ALL_RST_CFG_L);
-	regmap_writel(SW_ALL_RST_M_MSK, mmPSOC_GLOBAL_CONF_SW_ALL_RST_CFG_M);
-	regmap_writel(SW_ALL_RST_H_MSK, mmPSOC_GLOBAL_CONF_SW_ALL_RST_CFG_H);
-
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_SW_ALL_RST);
-}
-
-void hl_dev_reset(void)
-{
-	hl_pcie_disable_access();
-
-	greco_hw_fini();
-	wfe();
-}
diff --git a/arch/greco/arc_elf32/Makefile b/arch/greco/arc_elf32/Makefile
index 1482ec0..3a3cf82 100644
--- a/arch/greco/arc_elf32/Makefile
+++ b/arch/greco/arc_elf32/Makefile
@@ -2,5 +2,7 @@
 PPBOOT_MAX_SIZE := 16384
 
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/pll.c)
+PREBOOT_SRCS_C := $(filter-out $(HL_DEV_ARCH)/../common/reset.c, $(PREBOOT_SRCS_C))
+PPBOOT_SRCS_C := $(filter-out $(HL_DEV_ARCH)/../common/reset.c, $(PPBOOT_SRCS_C))
 
 image_validation: image_validation_common
diff --git a/arch/greco/arm/Makefile b/arch/greco/arm/Makefile
index 4f35394..8316644 100644
--- a/arch/greco/arm/Makefile
+++ b/arch/greco/arm/Makefile
@@ -2,7 +2,6 @@
 PPBOOT_MAX_SIZE := 18432
 PREBOOT_SRCS_C += $(shell ls lib/preboot/thermal_v2.c)
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/security.c)
-PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/reset.c)
 
 CFLAGS += -mstrict-align
 
diff --git a/arch/greco/arm/common.h b/arch/greco/arm/common.h
index 52dab79..be00d79 100644
--- a/arch/greco/arm/common.h
+++ b/arch/greco/arm/common.h
@@ -58,4 +58,8 @@
 
 #define FW_WORD_NUM		0x2000	/* FW_SIZE / 2 */
 
+#define SW_ALL_RST_L_MSK		0xC00FF8F2
+#define SW_ALL_RST_M_MSK		0xFFFFFFFF
+#define SW_ALL_RST_H_MSK		0x7FFFFF
+
 #endif /* COMMON_GRECO_H_ */
diff --git a/arch/greco/arm/reset.c b/arch/greco/arm/reset.c
deleted file mode 100644
index cd9e761..0000000
--- a/arch/greco/arm/reset.c
+++ /dev/null
@@ -1,41 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0
-/*
- * Copyright 2021 HabanaLabs, Ltd.
- * All Rights Reserved.
- */
-
-#include <psoc_global_conf_regs.h>
-#include <psoc_global_conf_masks.h>
-
-#include <bitops.h>
-#include <preboot.h>
-#include <pll.h>
-#include <uart.h>
-
-#include "common.h"
-
-#define SW_ALL_RST_L_MSK		0xC00FF8F2
-#define SW_ALL_RST_M_MSK		0xFFFFFFFF
-#define SW_ALL_RST_H_MSK		0x7FFFFF
-
-static void greco_hw_fini(void)
-{
-	/* Restart BTL/BLR upon hard-reset */
-	regmap_writel(0x770, mmPSOC_GLOBAL_CONF_BOOT_SEQ_RE_START);
-
-	udelay(1000);
-
-	regmap_writel(SW_ALL_RST_L_MSK, mmPSOC_GLOBAL_CONF_SW_ALL_RST_CFG_L);
-	regmap_writel(SW_ALL_RST_M_MSK, mmPSOC_GLOBAL_CONF_SW_ALL_RST_CFG_M);
-	regmap_writel(SW_ALL_RST_H_MSK, mmPSOC_GLOBAL_CONF_SW_ALL_RST_CFG_H);
-
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_SW_ALL_RST);
-}
-
-void hl_dev_reset(void)
-{
-	hl_pcie_disable_access();
-
-	greco_hw_fini();
-	wfe();
-}
diff --git a/arch/greco/common/reset.c b/arch/greco/common/reset.c
new file mode 100644
index 0000000..7f53d16
--- /dev/null
+++ b/arch/greco/common/reset.c
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright 2021 HabanaLabs, Ltd.
+ * All Rights Reserved.
+ */
+
+#include <psoc_global_conf_regs.h>
+#include <psoc_global_conf_masks.h>
+
+#include <bitops.h>
+#include <preboot.h>
+#include <pll.h>
+#include <uart.h>
+
+#include <common.h>
+
+
+static void greco_hw_fini(void)
+{
+	/* Restart BTL/BLR upon hard-reset */
+	regmap_writel(0x770, mmPSOC_GLOBAL_CONF_BOOT_SEQ_RE_START);
+
+	udelay(1000);
+
+	regmap_writel(SW_ALL_RST_L_MSK, mmPSOC_GLOBAL_CONF_SW_ALL_RST_CFG_L);
+	regmap_writel(SW_ALL_RST_M_MSK, mmPSOC_GLOBAL_CONF_SW_ALL_RST_CFG_M);
+	regmap_writel(SW_ALL_RST_H_MSK, mmPSOC_GLOBAL_CONF_SW_ALL_RST_CFG_H);
+
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_SW_ALL_RST);
+}
+
+void hl_dev_reset(void)
+{
+	hl_pcie_disable_access();
+
+	greco_hw_fini();
+	wfe();
+}
